aten/src/ATen/native/mkl/SparseBlasImpl.cpp

```diff
@@ -20,6 +22,40 @@ namespace mkl {
 
 namespace {
 
+c10::MaybeOwned<Tensor> prepare_dense_matrix_for_mkl(
+    const Tensor& tensor) {
+  if (tensor.is_non_overlapping_and_dense() ||
+      is_blas_compatible_row_major_order(tensor) ||
+      is_blas_compatible_column_major_order(tensor)) {
+    return c10::MaybeOwned<Tensor>::borrowed(tensor);
+  } else {
+    return c10::MaybeOwned<Tensor>::owned(
+        tensor.clone(at::MemoryFormat::Contiguous));
+  }
+}
+
+/*
+  Get row-major or column-major matrix.
+
+  Args:
+  * `tensor` - 2D strided Tensor.
+  * `row_major` - controls the memory layout.
+*/
+c10::MaybeOwned<Tensor> prepare_dense_matrix_for_mkl(
```

Comment:
Why do you need this overload and the one above? Is it for n-dim Tensor support?

Reply:
The one above returns either row or column-major matrix, it doesn't matter. This overload returns the matrix in a specified layout.